### PR TITLE
Minor cleanup to follow up 45062

### PR DESCRIPTION
**Pull request by ianatol**
Just some cleanup and partial addressing of review on https://github.com/JuliaLang/julia/pull/45062

### CODE REVIEW

File: base/compiler/ssair/passes.jl [hunk 1]
```diff
@@ -723,53 +723,45 @@ function perform_lifting!(compact::IncrementalCompact,
...
     if isa(vec, SimpleVector)
-        if valI <= length(val)
-            compact[idx] = vec[valI]
-        end
-        return
-    end
-
-    if isa(vec, SSAValue)
+        valI <= length(vec) || return
+        compact[idx] = QuoteNode(vec[valI])
```
Comment (ianatol) [inline base/compiler/ssair/passes.jl:737]: @vtjnash I don't quite understand why this needs `QuoteNode` (per https://github.com/JuliaLang/julia/pull/45062#discussion_r968884654). It's probably really simple, but could you explain?

**Comment by ianatol** [comment]
In addition, I'd like to make `_lift_svec_ref` more understandable and general, but that will have to wait for now

File: base/compiler/ssair/passes.jl (hunk 1 above)
Comment (vtjnash) [reply base/compiler/ssair/passes.jl:737]: It may need a quote node since the right side is an arbitrary value and the left side expects an instruction

**Review by Keno — APPROVED** [review]

File: base/compiler/ssair/passes.jl (hunk 1 above)
Comment (Keno) [reply base/compiler/ssair/passes.jl:737]: usually we use `quoted` here rather than `QuoteNode` to avoid quoting things that don't need actual QuotNodes

**Pull request merged**